source/common/common/key_value_store_base.cc

```diff
@@ -70,7 +93,6 @@ void KeyValueStoreBase::addOrUpdate(absl::string_view key_view, absl::string_vie
   std::string value(value_view);
   // Do not add if ttl is <= 0
   if (ttl && ttl <= std::chrono::seconds(0)) {
-    ASSERT(false);
```

Comment:
Previously, TTL<0 represented a malformed input since there's no good reason e.g. traffic director would send a negative TTL (I think).  But this update actually uses negative TTLs here, if a TTL expires before the store gets reloaded from disk.

Reply:
I think if the TTL has expired, we shouldn't reload from disk, i.e. we can not call addOrUpdate if the ttl in https://github.com/envoyproxy/envoy/pull/23318/files/537f0b1df57b9a7fc03898dca95d653db241d739#diff-b9ac1e7fb952c169f6ee95b58cb15c65ac31c8e8b421f521c13546e6b511fd8eR76 is a negative value

Reply:
Yeah.  But the way it works out as written is that we call `addOrUpdate` with a negative TTL, and then hit this `if` statement and return without doing anything.

The alternative would be to put the logic enforcing ttl>0 in two different functions which seems worse to me.

Reply:
In `addOrUpdate`, you could just keep the assert (or log a warning, or something).  That way, we know if we passed in something with a negative TTL, which should be an error.

Reply:
Ok I will return it to the previous format.